UnitTests/StreamCopierTest.cs

```diff
@@ -41,96 +40,103 @@ public void SetUp ()
 
-      inputMock.Stub (s => s.Length).Return (streamLength);
-      inputMock.Stub (s => s.CanSeek).Return (true);
-      SetupReadExpectation (inputMock, Enumerable.Range (0, contentLength).Select (i => (byte) i));
-      SetupReadExpectation (inputMock, new byte[] { });
+      var sequence = new MockSequence();
+      inputMock.SetupGet (s => s.Length).Returns (streamLength);
+      inputMock.SetupGet (s => s.CanSeek).Returns (true);
+      SetupReadExpectation (inputMock, sequence,Enumerable.Range (0, contentLength).Select (i => (byte) i));
+      SetupReadExpectation (inputMock, sequence,new byte[] { });
 
       MemoryStream outputStream = new MemoryStream();
 
-      Assert.That (_streamCopier.CopyStream (inputMock, outputStream), Is.False);
+      Assert.That (_streamCopier.CopyStream (inputMock.Object, outputStream), Is.False);
 
-      inputMock.VerifyAllExpectations();
+      inputMock.Verify();
     }
 
     [Test]
     public void CopyStream_LengthDiffersWithContentTooLong_ThrowsIOException ()
     {
-      Stream inputMock = MockRepository.GenerateStrictMock<Stream>();
+      var inputMock = new Mock<Stream>();
       const int streamLength = 10;
       const int contentLength = streamLength * 2;
 
-      inputMock.Stub (s => s.Length).Return (streamLength);
-      inputMock.Stub (s => s.CanSeek).Return (true);
-      SetupReadExpectation (inputMock, Enumerable.Range (0, contentLength).Select (i => (byte) i));
-      SetupReadExpectation (inputMock, new byte[] { });
+      var sequence = new MockSequence();
+      inputMock.SetupGet (s => s.Length).Returns (streamLength);
+      inputMock.SetupGet (s => s.CanSeek).Returns (true);
+      SetupReadExpectation (inputMock, sequence,Enumerable.Range (0, contentLength).Select (i => (byte) i));
+      SetupReadExpectation (inputMock, sequence,new byte[] { });
 
       MemoryStream outputStream = new MemoryStream();
 
       Assert.That (
-          () => _streamCopier.CopyStream (inputMock, outputStream),
+          () => _streamCopier.CopyStream (inputMock.Object, outputStream),
           Throws.TypeOf<IOException>().With.Message.EqualTo ("The stream returned more data than the stream length (10 bytes) specified."));
 
-      inputMock.VerifyAllExpectations();
+      inputMock.Verify();
     }
 
     [Test]
     public void CopyStream_UserCancels_ReturnsFalse ()
     {
-      Stream inputMock = MockRepository.GenerateStrictMock<Stream>();
+      var inputMock = new Mock<Stream>();
 
       const int numberOfNotificationsToWait = 5;
 
-      inputMock.Expect (s => s.Read (null, 0, 0)).IgnoreArguments().Return (1).Repeat.Times (numberOfNotificationsToWait);
-      inputMock.Stub (s => s.Length).Return (10);
+      var sequence = new MockSequence();
+      for (int i = 0; i < numberOfNotificationsToWait; i++)
+      {
+        inputMock.InSequence (sequence).Setup (s => s.Read (It.IsAny<Byte[]>(), It.IsAny<int>(), It.IsAny<int>())).Returns (1).Verifiable();
+      }
```

Comment:
I assume you did it this way to keep the expectation "where it was before" but I would still suggest just doing a simple setup and verifying `Times` at the end of the test, as it seems to be the Moq way of doing things